packages/cli/src/processed-data-managers/processed-data-helper.ts

```diff
@@ -0,0 +1,322 @@
+	}
+
+	private static compareValues(
+		mode: ProcessedDataMode,
+		value1: ProcessedDataItemTypes,
+		value2: ProcessedDataItemTypes,
+	): boolean {
+		if (mode === 'latestIncrementalKey') {
+			const num1 = Number(value1);
+			const num2 = Number(value2);
+			if (!isNaN(num1) && !isNaN(num2)) {
+				return num1 > num2;
+			}
+			throw new ApplicationError(
+				'Invalid value. Only numbers are supported in mode "latestIncrementalKey"',
+			);
+		} else if (mode === 'latestDate') {
+			const date1 = new Date(value1 as string);
+			const date2 = new Date(value2 as string);
+
+			if (!isNaN(date1.getTime()) && !isNaN(date2.getTime())) {
+				return date1 > date2;
+			} else {
+				throw new ApplicationError(
+					'Invalid value. Only valid dates are supported in mode "latestDate"',
+				);
+			}
+		} else {
+			throw new ApplicationError(
+				"Invalid mode. Only 'latestIncrementalKey' and 'latestDate' are supported.",
+			);
+		}
+	}
+
+	private static createContext(
+		context: ProcessedDataContext,
+		contextData: ICheckProcessedContextData,
+	): string {
+		if (context === 'node') {
+			console.log('contextData.node', contextData, contextData.node);
+			if (!contextData.node) {
+				throw new ApplicationError(
+					"No node information has been provided and can so not use context 'node'",
+				);
+			}
+			// Use the node ID to make sure that the data can still be accessed and does not get deleted
+			// whenver the node gets renamed
+			return `n:${contextData.node.id}`;
+		}
+
+		return '';
+	}
+
+	private static createValueHash(value: ProcessedDataItemTypes): string {
+		return createHash('md5').update(value.toString()).digest('base64');
+	}
+
+	async checkProcessed(
+		items: ProcessedDataItemTypes[],
+		context: ProcessedDataContext,
+		contextData: ICheckProcessedContextData,
+		options: ICheckProcessedOptions,
+	): Promise<ICheckProcessedOutput> {
+		const returnData: ICheckProcessedOutput = {
+			new: [],
+			processed: [],
+		};
+
+		const processedData = await Container.get(ProcessedDataRepository).findOne({
+			where: {
+				workflowId: contextData.workflow.id as string,
+				context: ProcessedDataHelper.createContext(context, contextData),
+			},
+		});
+
+		if (processedData && processedData.value.mode !== options.mode) {
+			throw new ApplicationError(
+				'Deduplication data was originally saved with an incompatible setting of the ‘Keep Items Where’ parameter. ’Try Clean Database’ operation to reset.',
+			);
+		}
+
+		if (!processedData) {
+			// If there is nothing it the database all items are new
+			returnData.new = items;
+			return returnData;
+		}
+
+		if (['latestIncrementalKey', 'latestDate'].includes(options.mode)) {
+			const processedDataValue = processedData.value as IProcessedDataLatest;
+
+			const incomingItems = ProcessedDataHelper.sortEntries(items, options.mode);
+			incomingItems.forEach((item) => {
+				if (ProcessedDataHelper.compareValues(options.mode, item, processedDataValue.data)) {
```

Comment:
Is this logic correct? `compareValues` check if `a` is greater than `b`. It does not check for equality. Maybe the `compareValues` could return `1 | -1 | 0` and we could check `=== 0` here? That way it works out of the box for `.sort()` method as well

Reply:
in this mode we compare against the last stored date or id and we only consider greater values as none duplicate, I could change the compareValues function but the only value we are interested in will be 1 when a>b